lib/temporal/workflow/context.rb

```diff
@@ -287,6 +287,14 @@ def on_signal(&block)
         end
       end
 
+      def on_query(query, &block)
```

Comment:
I wonder if we should allow a "catch-all" query handler via `workflow.on_query do |query, input|` (without specifying a query). What do you think?

Reply:
Yeah I agree, hopefully we can give both signals and queries similar flexibility of handlers. Queries probably need a preference in which one is elected to be used - or warnings / errors if attempting to use both forms in the same workflow, potentially.

Reply:
Hmm… I have a sense that we can support both forms with the same API. I don't think there's too much room for confusion there:

```ruby
workflow.on_query('query') do |input|
```

and

```ruby
workflow.on_query do |query, input|
```

we also should be able to validate by checking block's arity when called

Reply:
Or do you think there might be confusion with `input`'s optionality?

Reply:
I sweeped through and added the catch-all query handler support in latest round of changes (within the same API). This bit of code should be outdated now.